docs/proposals/20230612-node-os-image-caching.md

```diff
@@ -119,29 +120,33 @@ None currently known to be needed
 #### Non-Functional Requirements
 
 ##### NFR1
-The solution should increase the performance of horizontal scaling over time as people use this feature
 
 ##### NFR2
-The solution should cause no increased risks with security or reliability
 
 ### Implementation Details/Notes/Constraints
 
-The plan is to create a Controller which implements the feature with the Node Prototype Pattern as desired.
+The plan is to modify the existing Controllers with the Node Prototype Pattern as desired. These controller additions can be added to AzureMachine Controller and AzureMachinePool Controller.
+
+An operator will be able to decide to turn the feature on or off with an environment variable on creation of a cluster, and then can update the AzureMachineTemplate and AzureMachinePool to customize how long they want the caching interval to be (see the yaml files below in this section for the caching interval).
```

Comment:
> with an environment variable on creation of a cluster

Similar features in CAPZ have been put behind feature flags like the ones defined here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/af8668a62483d7484792b4767bc4089f1f96d584/feature/feature.go#L56-L63

Then those are enabled on the capz-controller-manager Pod in the Deployment spec defined here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/af8668a62483d7484792b4767bc4089f1f96d584/config/manager/manager.yaml#L26

So generally, users would enable features in CAPZ like this:
```
export MY_FAVORITE_FEATURE=true
clusterctl init -i azure
```

That makes the feature available for all clusters managed by that CAPZ instance. To make this more granular on a per-cluster or per-machine pool basis, I wonder if we could use the presence or absence of node caching-specific configuration (like the time interval) to ultimately determine whether or not the node caching will be invoked.

Reply:
Yup and the sync about this was very helpful, thank you so much!